urdf_model/include/urdf_model/utils.h

```diff
@@ -62,6 +65,28 @@ void split_string(std::vector<std::string> &result,
   }
 }
 
+// This is a locale-safe version of string-to-double, which is suprisingly
+// difficult to do correctly.  This function ensures that the C locale is used
+// for parsing, as that matches up with what the XSD for double specifies.
+// On success, the double is returned; on failure, a std::runtime_error is
+// thrown.
+static inline double strToDouble(const char *in)
```

Comment:
Why use `const char *` instead of `const std::string &`? Just curious, not asking you to change it, just hoping to learn something.

Reply:
There is a good (but sneaky) reason for it.  I'm hoping that once we get this basic support in, we can use this more generally throughout the urdfdom libraries to avoid this problem.  As one example, https://github.com/ros/urdfdom/blob/master/urdf_parser/src/link.cpp#L120 probably suffers from a similar problem (as do several other std::stod conversions in there).  Several of those get a `const char *` from tinyxml, so I figured this interface was more generic rather than having to wrap those inside strings.

Reply:
Oh, right, I forgot I also did another PR to actually fix all of that as well: https://github.com/ros/urdfdom/pull/105